### PR TITLE
Cu 86b0p7m2p add a UI message when a search result return no results jose

### PR DESCRIPTION
**Pull request by josewebdev2000**
Now when the user searches for movie titles and none appear. The user will see a "No Results Found" message that will inspire him/her to look for other titles.

### CODE REVIEW

**Comment by allantito** [comment]
Task linked: [CU-86b0p7m2p Add a UI message when a search result return no results](https://app.clickup.com/t/86b0p7m2p)

File: P1/Project_Code/frontend/src/routes/SearchResults/SearchResults.route.jsx [hunk 1]
```diff
@@ -8,15 +8,27 @@ export const SearchResults = () => {
...
-                filteredMovies.map((movie, index) => (
-                    <Grid item key={index}>
-                        <MediaCard movie={movie} type="searched"/>
-                    </Grid>
-                ))
-            }
-        </Grid>
-    )
+    // Show movies searched in case there were any
+    if (filteredMovies.length > 0)
```
Comment (ethanfreire) [inline P1/Project_Code/frontend/src/routes/SearchResults/SearchResults.route.jsx:12]: Suggest refactoring this code to include a check that filteredMovies is not null or undefined . See example below 
```
// Check if filteredMovies is defined and not null
 if (filteredMovies && filteredMovies.length > 0) {
some code here
}
// else if no movies were filtered or filteredMovies is undefined/null, display "No Results Found" message
else {
some code here
}
```

Comment (ethanfreire) [reply P1/Project_Code/frontend/src/routes/SearchResults/SearchResults.route.jsx:12]: The condition `if (filteredMovies && filteredMovies.length > 0)` checks if `filteredMovies` is not `undefined` or `null` before checking its `length`. This prevents potential runtime errors when trying to access the length property of an undefined or null value.

**Review by ethanfreire — COMMENTED** [review]
I've left some feedback on refactoring and implementing a check on filteredMovies value

**Review by ethanfreire — COMMENTED** [review]
Looks good to me just review messages left on your other pull requests.

File: P1/Project_Code/frontend/src/components/Layout/Layout.jsx [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
+  const navigate = useNavigate();
+
+  // Grab Global User Data
+  const { user } = useContext(UserContext);
+
+  // Redirect to Login Route if no user
+  useEffect(() => {
+    if (!user)
+    {
+      navigate("/login");
```
Comment (ryanRATM) [inline P1/Project_Code/frontend/src/components/Layout/Layout.jsx:21]: recommend moving this check logged in to own class and place in router logic
https://blog.logrocket.com/authentication-react-router-v6/
https://www.robinwieruch.de/react-router-authentication/

**Pull request merged**